extensions-contrib/opentsdb-emitter/src/main/java/org/apache/druid/emitter/opentsdb/EventConverter.java

```diff
@@ -41,17 +41,24 @@
   private static final Pattern WHITESPACE = Pattern.compile("[\\s]+");
 
   private final Map<String, Set<String>> metricMap;
+  private final String namespacePrefix;
 
-  public EventConverter(ObjectMapper mapper, String metricMapPath)
+  public EventConverter(ObjectMapper mapper, String metricMapPath, String namespacePrefix)
   {
     metricMap = readMap(mapper, metricMapPath);
+    this.namespacePrefix = namespacePrefix;
   }
 
   protected String sanitize(String metric)
   {
     return WHITESPACE.matcher(metric.trim()).replaceAll("_").replace('/', '.');
   }
 
+  private String buildNamespace()
```

Comment:
nit: I think this might be a fair bit cleaner to rework this method to take the metric name and do the call to `sanitize` in here, maybe something like:

```
private String buildMetric(String metric)
{
  final String sanitized = sanitize(metric);
  if (Strings.isNullOrEmpty(namespacePrefix)) {
    return sanitized;
  } else {
    return StringUtils.format("%s.%s", namespacePrefix, sanitized);
  }  
}
```

then creating the `OpentsdbEvent` can just be

```
    ...
    return new OpentsdbEvent(buildMetric(metric), timestamp, value, tags);
```

Reply:
Makes perfect sense, i've added  `buildMetric` more or like what you suggested.